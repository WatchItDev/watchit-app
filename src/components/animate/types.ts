// ----------------------------------------------------------------------

type EaseType =
  | "linear"
  | "easeIn"
  | "easeOut"
  | "easeInOut"
  | "circIn"
  | "circOut"
  | "circInOut"
  | "backIn"
  | "backOut"
  | "backInOut"
  | "anticipate"
  | number[];

export interface VariantsType {
  distance?: number;
  durationIn?: number;
  durationOut?: number;
  easeIn?: EaseType;
  easeOut?: EaseType;
}

export interface TranHoverType {
  duration?: number;
  ease?: EaseType;
}

export interface TranEnterType {
  durationIn?: number;
  easeIn?: EaseType;
}

export interface TranExitType {
  durationOut?: number;
  easeOut?: EaseType;
}

export interface BackgroundType {
  colors?: string[];
  duration?: number;
  ease?: EaseType;
}
