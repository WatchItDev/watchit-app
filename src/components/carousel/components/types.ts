import {BoxProps} from "@mui/material/Box";
import {SxProps, Theme} from "@mui/material/styles";
import {IconifyProps} from "@src/components/iconify";
import {StackProps} from "@mui/material/Stack";
import React from "react";
import {IconButtonProps} from "@mui/material/IconButton";

export interface CarouselDotsProps extends BoxProps {
  rounded?: boolean;
  sx?: SxProps<Theme>;
}

export type ArrowIconProps = {
  icon?: IconifyProps; // Right icon
  isRTL?: boolean;
};

export type CarouselArrowIndexProps = {
  index: number;
  total: number;
  icon?: IconifyProps; // Right icon
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
  sx?: SxProps<Theme>;
};

export interface CarouselArrowsProps extends StackProps {
  shape?: 'circular' | 'rounded';
  filled?: boolean;
  children?: React.ReactNode;
  icon?: IconifyProps; // Right icon
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
  leftButtonProps?: IconButtonProps;
  rightButtonProps?: IconButtonProps;
}

export interface CarouselArrowsStyledIconButtonProps extends IconButtonProps {
  filled?: boolean;
  shape?: 'circular' | 'rounded';
  hasChild?: boolean;
}

export interface NavigationArrowsProps {
  next: () => void;
  prev: () => void;
}

export type CarouselDotsStyledRootProps = {
  rounded: boolean;
};

