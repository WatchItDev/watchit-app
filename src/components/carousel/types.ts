import React from "react";

import {Profile, VideoMetadataV3} from "@lens-protocol/api-bindings";
// @ts-ignore
import { Post,PublicationMetadataMediaVideoFieldPolicy } from "@lens-protocol/api-bindings/dist/declarations/src/lens/graphql/generated";

import {StackProps} from "@mui/material/Stack";
import {IconButtonProps} from "@mui/material/IconButton";

import {BoxProps} from "@mui/material/Box";
import {SxProps, Theme} from "@mui/material/styles";
import {IconifyProps} from "@src/components/iconify";


export interface CarouselDotsProps extends BoxProps {
  rounded?: boolean;
  sx?: SxProps<Theme>;
}

interface CarouselNavigationActions {
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
}

export type ArrowIconProps = {
  icon?: IconifyProps;
  isRTL?: boolean;
};

export type CarouselArrowIndexProps = CarouselNavigationActions & {
  index: number;
  total: number;
  icon?: IconifyProps;
  sx?: SxProps<Theme>;
};

export interface CarouselArrowsProps extends StackProps, CarouselNavigationActions {
  shape?: 'circular' | 'rounded';
  filled?: boolean;
  children?: React.ReactNode;
  icon?: IconifyProps; // Right icon
  leftButtonProps?: IconButtonProps;
  rightButtonProps?: IconButtonProps;
}

export interface CarouselArrowsStyledIconButtonProps extends IconButtonProps {
  filled?: boolean;
  shape?: 'circular' | 'rounded';
  hasChild?: boolean;
}

export interface NavigationArrowsProps {
  next: () => void;
  prev: () => void;
}

export type CarouselDotsStyledRootProps = {
  rounded: boolean;
};

interface CarouselResizable {
  minItemWidth: number;
  maxItemWidth: number;
}

export interface CarouselWrapperProps<T> extends CarouselResizable{
  data: T[];
  title?: string;
  renderSlide: (slideItems: T[], itemsPerRow: number, index: number) => React.ReactNode;
  carouselSettings: any;
  boxStyle?: any;
  boxClassName?: string;
}

export interface CarouselSlideProps<T> {
  items: T[];
  itemsPerRow: number;
  renderItem: (item: T) => React.ReactNode;
}

export type CarouselCreatorsProps = CarouselResizable & {
  data: Profile[];
  title?: string;
};

export type CarouselPosterMiniProps = CarouselResizable & {
  data: Post[];
  title?: string;
};

export type CarouselTopTitlesProps = {
  posts: Post[];
  category?: string;
};

export type CarouselPosterSlideProps = {
  items: Post[];
  itemsPerRow: number;
};

export interface PublicationType extends Post {
  altTag: string;
  metadata: VideoMetadataV3 | PublicationMetadataMediaVideoFieldPolicy;
}
