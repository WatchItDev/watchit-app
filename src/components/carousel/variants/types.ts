import {Profile} from "@lens-protocol/api-bindings";
// @ts-ignore
import type {Post} from "@lens-protocol/api-bindings/dist/declarations/src/lens/graphql/generated";
import React from "react";

export interface CarouselWrapperProps<T> {
  data: T[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
  renderSlide: (slideItems: T[], itemsPerRow: number, index: number) => React.ReactNode;
  carouselSettings: any;
  boxStyle?: any;
  boxClassName?: string;
}

export type CarouselSlideProps = {
  items: Profile[];
  itemsPerRow: number;
};

export type CarouselCreatorsProps = {
  data: Profile[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
};

export type CarouselPosterMiniProps = {
  data: Post[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
};

export type CarouselTopTitlesProps = {
  posts: Post[];
  category?: string;
};

export type CarouselPosterSlideProps = {
  items: Post[];
  itemsPerRow: number;
};
