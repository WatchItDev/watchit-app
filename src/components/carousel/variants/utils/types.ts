export interface calculateItemsPerSlideProps {
  parentWidth: number;
  minItemWidth: number;
  maxItemWidth: number;
}
