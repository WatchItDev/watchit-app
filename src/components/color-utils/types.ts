// @mui
import { StackProps } from '@mui/material/Stack';

// ----------------------------------------------------------------------

export interface ColorPreviewProps extends StackProps {
  colors: string[];
  limit?: number;
}

export interface ColorPickerProps extends StackProps {
  multi?: boolean;
  colors: string[];
  selected: string | string[];
  limit?: 'auto' | number;
  onSelectColor: (color: string | string[]) => void;
}
