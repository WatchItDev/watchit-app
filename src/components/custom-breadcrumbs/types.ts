// @mui
import { BreadcrumbsProps } from '@mui/material/Breadcrumbs';

// ----------------------------------------------------------------------

export type BreadcrumbsLinkProps = {
  name?: string;
  href?: string;
  icon?: React.ReactElement;
};

export interface CustomBreadcrumbsProps extends BreadcrumbsProps {
  heading?: string;
  moreLink?: string[];
  activeLast?: boolean;
  action?: React.ReactNode;
  links: BreadcrumbsLinkProps[];
}
