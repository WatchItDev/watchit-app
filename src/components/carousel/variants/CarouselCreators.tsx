import { Profile } from '@lens-protocol/api-bindings';
import CarouselWrapper from './CarouselWrapper';
import CarouselSlide from '@src/components/carousel/components/CarouselSlide';
import { UserItem } from '@src/components/user-item';
import { CarouselCreatorsProps } from '../types';
import {PublicationType} from "@lens-protocol/react-web";

export default function CarouselCreators(params: CarouselCreatorsProps) {
  const { data, title, minItemWidth, maxItemWidth } = params;

  const carouselSettings = {
    infinite: false,
    slidesToShow: 1,
    speed: 500,
    rows: 1,
    slidesPerRow: 1,
    lazyLoad: 'progressive',
  };

  const boxStyle = {
    '.slick-track': {
      height: '100%',
    },
    '.slick-slide': {
      height: '100%',
    },
    '.slick-slide > div': {
      height: '100%',
    },
  };

  const renderItem = (profile: Profile) => (
    <UserItem profile={profile} onActionFinished={() => {}} followButtonMinWidth={90} />
  );

  const carouselParams = {
    data,
    title,
    minItemWidth,
    maxItemWidth,
    renderSlide: (slideItems: PublicationType, itemsPerRow: number, index: number) => (
      <CarouselSlide
        key={`slide-${index}`}
        items={slideItems}
        itemsPerRow={itemsPerRow}
        renderItem={renderItem}
      />
    ),
    carouselSettings,
    boxStyle,
  }

  return (
    <CarouselWrapper{... carouselParams} />
  );
}
