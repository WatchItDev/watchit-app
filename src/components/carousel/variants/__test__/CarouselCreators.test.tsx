import '../../../../../__mocks__/web3auth';
import { describe, it, expect } from 'vitest';
import CarouselCreators from '@src/components/carousel/variants/carousel-creators.tsx';
import { CarouselCreatorsProps } from '@src/components/carousel/types.ts';
import { renderWithStoreAndRouter } from '@src/utils/testing/Testing.tsx';

describe('[COMPONENTS]: CarouselCreators', () => {
  const mockData = [
    {
      id: '1',
      metadata: {
        picture: null,
        __typename: 'ProfileMetadata',
        bio: null,
        rawURI: 'https://watchit.movie',
        appId: null,
        coverPicture: null,
        displayName: 'Test User 1',
        attributes: null,
      },
      __typename: 'Profile',
      invitedBy: null,
    },
  ];

  const defaultProps: CarouselCreatorsProps = {
    data: mockData,
    title: 'Test Carousel',
    minItemWidth: 100,
    maxItemWidth: 200,
  };

  it('to match snapshot', () => {
    expect(
      renderWithStoreAndRouter(<CarouselCreators {...defaultProps} />)
        .baseElement,
    ).toMatchSnapshot();
  });

  it('renders the correct number of slides', () => {
    const slides = renderWithStoreAndRouter(
      <CarouselCreators {...defaultProps} />,
    ).container.querySelectorAll('.slick-slider');
    expect(slides.length).toBe(mockData.length);
  });

  it('applies the correct carousel settings', () => {
    const slickTrack = renderWithStoreAndRouter(
      <CarouselCreators {...defaultProps} />,
    ).container.querySelector('.slick-track');
    expect(slickTrack).toHaveStyle('height: 100%');
  });

  it('renders the title correctly', () => {
    expect(
      renderWithStoreAndRouter(
        <CarouselCreators {...defaultProps} />,
      ).getByText('Test Carousel'),
    ).toBeInTheDocument();
  });

  it('handles empty data gracefully', () => {
    const emptyProps = { ...defaultProps, data: [] };
    const slides = renderWithStoreAndRouter(
      <CarouselCreators {...emptyProps} />,
    ).container.querySelectorAll('.slick-slide');
    expect(slides.length).toBe(0);
  });
});
