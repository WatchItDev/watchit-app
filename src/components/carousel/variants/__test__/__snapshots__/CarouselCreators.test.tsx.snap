// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS]: CarouselCreators > to match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardHeader-root css-12jpqf3-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
          >
            Test Carousel
          </span>
        </div>
        <div
          class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="leftButton"
            tabindex="0"
            type="button"
          >
            <svg
              class="tabler-icon tabler-icon-chevron-left "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 6l-6 6l6 6"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="rightButton"
            tabindex="0"
            type="button"
          >
            <svg
              class="tabler-icon tabler-icon-chevron-right "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 6l6 6l-6 6"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-10jug1b-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-8lmyt0"
        >
          <div
            class="slick-slider slick-initialized"
          >
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <div
                        class="MuiBox-root css-k008qs"
                      >
                        <div
                          class="MuiBox-root css-rmtq2p"
                        >
                          <div
                            class="MuiBox-root css-urjiuj"
                          >
                            <span
                              class="component-image MuiBox-root css-1w0a4u"
                            >
                              <span
                                class="component-image-wrapper lazy-load-image-background blur"
                                style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                              >
                                <img
                                  alt="Profile Cover"
                                  class="MuiBox-root css-3j6ntj"
                                  src="https://picsum.photos/seed/1/1920/820"
                                />
                              </span>
                            </span>
                            <div
                              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vihmqx-MuiPaper-root-MuiCard-root"
                            >
                              <div
                                class="MuiAvatar-root MuiAvatar-rounded css-13ubyhz-MuiAvatar-root"
                              >
                                <img
                                  alt=""
                                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                                  src="https://api.dicebear.com/9.x/bottts-neutral/svg?seed=1"
                                />
                              </div>
                              <div
                                class="MuiBox-root css-7gc5b8"
                              >
                                <div
                                  class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
                                >
                                  <span
                                    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap MuiListItemText-primary css-f69c18-MuiTypography-root"
                                  >
                                    <div
                                      class="MuiBox-root css-geek62"
                                    >
                                      <p
                                        class="MuiTypography-root MuiTypography-body1 css-nl8q6o-MuiTypography-root"
                                      />
                                      <div
                                        class="MuiBox-root css-1wqjpco"
                                      />
                                    </div>
                                  </span>
                                  <span
                                    class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap MuiListItemText-secondary css-uhfo76-MuiTypography-root"
                                  >
                                    1
                                  </span>
                                </div>
                                <div
                                  class="MuiBox-root css-0"
                                >
                                  <button
                                    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1hhktcy-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                                    id=":r0:"
                                    tabindex="0"
                                    title="Unfollow"
                                    type="button"
                                  >
                                    Unfollow
                                    <span
                                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                    />
                                  </button>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-k008qs"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
